test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -437,22 +440,23 @@ public void HubsCannotHaveOverloadedMethods()
             }
         }
 
-        [Fact]
-        public async Task BroadcastHubMethod_SendsToAllClients()
+        [Theory]
+        [MemberData(nameof(HubTypes))]
+        public async Task BroadcastHubMethodSendsToAllClients(Type hubType)
         {
             var serviceProvider = CreateServiceProvider();
 
-            var endPoint = serviceProvider.GetService<HubEndPoint<MethodHub>>();
+            dynamic endPoint = serviceProvider.GetService(GetEndPointType(hubType));
 
             using (var firstClient = new TestClient())
             using (var secondClient = new TestClient())
             {
-                var firstEndPointTask = endPoint.OnConnectedAsync(firstClient.Connection);
-                var secondEndPointTask = endPoint.OnConnectedAsync(secondClient.Connection);
+                Task firstEndPointTask = endPoint.OnConnectedAsync(firstClient.Connection);
+                Task secondEndPointTask = endPoint.OnConnectedAsync(secondClient.Connection);
 
                 await Task.WhenAll(firstClient.Connected, secondClient.Connected).OrTimeout();
 
-                await firstClient.SendInvocationAsync(nameof(MethodHub.BroadcastMethod), "test").OrTimeout();
+                await firstClient.SendInvocationAsync("BroadcastMethod", "test").OrTimeout();
```

Comment:
Leaving nameof would be misleading. The idea is that all hubs have the same "shape" (i.e. methods and parameters) and nameof() of one hub to use as the name of the method of a different hub is gross.